rustls/src/hash_hs.rs

```diff
@@ -129,11 +123,14 @@ impl HandshakeHash {
     /// 'handshake_hash' handshake message.  Start this hash
     /// again, with that message at the front.
     pub fn rollup_for_hrr(&mut self) {
-        let old_hash = self.ctx.take().unwrap().finish();
+        let ctx = self.ctx.as_mut().unwrap();
+
+        let old_ctx = mem::replace(ctx, digest::Context::new(ctx.algorithm()));
+        let old_hash = old_ctx.finish();
         let old_handshake_hash_msg =
             HandshakeMessagePayload::build_handshake_hash(old_hash.as_ref());
 
-        self.ctx = Some(digest::Context::new(self.alg.unwrap()));
+        self.ctx = Some(digest::Context::new(ctx.algorithm()));
```

Comment:
It seems wrong that we call `Context::new` twice now.

Reply:
Yup, that is an oversight. Will fix it soon.

Reply:
Fixed.